text/0021-enable-profiling-in-sentry-cli.md

```diff
@@ -0,0 +1,53 @@
+* Start Date: 2022-09-29
+* RFC Type: feature
+* RFC PR: [#21](https://github.com/getsentry/rfcs/pull/21)
+
+# Summary
+
+This RFC details the proposal to add profiling instrumentation to the `sentry CLI`
+
+# Motivation
+
+We've recently added the support for profiling in the Rust SDK and we've been looking for a Sentry tool that we could use to dogfood it.
```

Comment:
Maybe start by enabling this conditionally on SDKs that use sentry-cli in CI, i.e. check that `env['GITHUB_REPOSITORY_OWNER'] == 'getsentry'`?

Reply:
@vaind good point! that would also be an option. 

Do we have a rough estimate of `getsentry` projects using `sentry-cli` in CI to figure out if and how much data we could expect to get?

One drawback I can think of is that probably most of the runner used in CI will be Linux machines so we won't get to test it on Intel Mac or M1 Mac.

Reply:
`sentry-unity` integration tests run on the three platforms, still no arm mac, of course.

No idea about the other project usage though, IIRC it should be used by the android gradle plugin & fastlane plugin

Reply:
@mitsuhiko @kamilogorek what do you think? it seems like this could be a viable option that would limit the risk for customers.

Reply:
How would that be any different from making it generally opt-in?

This way instead of `env['GITHUB_REPOSITORY_OWNER'] == 'getsentry'` you could do `env['SENRY_CLI_PROFILING'] == '1'` in chosen `getsentry` repositories.

You'd need to do that per repo, but this way you have also more granular control over which repos would run it, as well as allow for non-sentry repos to opt-in. We'd also not have any magic values in the cli itself, as `== 'getsentry'` would be stuck there forever in the binary.

Reply:
> How would that be any different from making it generally opt-in?

Exactly by having to do the opt-in (PR, review) in every repo.

> We'd also not have any magic values in the cli itself, as == 'getsentry' would be stuck there forever in the binary.

My idea was to run this for a short while before doing the original proposal of this PR, i.e. enable it by default at which point the check of the repo owner can be removed.